metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -221,7 +229,11 @@ def get_databases(self, db_name: str) -> List[SnowflakeDatabase]:
                 last_altered=database["LAST_ALTERED"],
                 comment=database["COMMENT"],
             )
-            databases.append(snowflake_db)
+            self.report.report_entity_scanned(snowflake_db.name, "database")
+            if not self.config.database_pattern.allowed(snowflake_db.name):
+                self.report.report_dropped(f"{snowflake_db.name}.*")
+            else:
+                databases.append(snowflake_db)
```

Comment:
I moved it inside to avoid pattern checking at multiple places., and I didn't wish to add another iterator over the list in snowflake source. Your reasoning makes sense however and I'll move this filtering inside a wrapper method in SnowflakeSource.